gms/gossiper.cc

```diff
@@ -427,7 +428,7 @@ future<> gossiper::handle_shutdown_msg(inet_address from, std::optional<int64_t>
         co_return;
     }
 
-    auto permit = co_await this->lock_endpoint(from);
+    auto permit = co_await this->lock_endpoint(from, null_permit_id);
```

Comment:
Why not have a version that defaults to use null_permit_id the user just calls:

this->lock_endpoint(from)

Reply:
It enforces the caller to make a conscious choice about the permit_id.
Passing a null one on a nested call may cause deadlock.

Reply:
The user can still pass a null id inside a nested call. This does not really enforce anything.